internal/fnruntime/runner.go

```diff
@@ -174,6 +173,50 @@ func parseStructuredResult(yml *yaml.RNode, fnResult *fnresult.Result) error {
 	if err != nil {
 		return err
 	}
+
+	return parseNameAndNamespace(yml, fnResult)
+}
+
+func parseNameAndNamespace(yml *yaml.RNode, fnResult *fnresult.Result) error {
+	items, err := yml.Elements()
+	if err != nil {
+		return err
+	}
+
+	for i := range items {
+		if err := getResourceRefMetadata(items[i], fnResult, i); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func getResourceRefMetadata(item *yaml.RNode, fnResult *fnresult.Result, i int) error {
```

Comment:
we don't need the `i` arg.

Better name for the function would be: `populateResourceRef`

Reply:
without `i`, how will we know which fnResult item this corresponds to? It's used on line 217